core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -352,8 +353,20 @@ private[spark] class CoarseGrainedExecutorBackend(
                 // We can only trust allBlocksMigrated boolean value if there were no tasks running
                 // since the start of computing it.
                 if (allBlocksMigrated && (migrationTime > lastTaskRunningTime)) {
-                  logInfo("No running tasks, all blocks migrated, stopping.")
-                  exitExecutor(0, ExecutorLossMessage.decommissionFinished, notifyDriver = true)
+                  val pendingFetches = env.blockManager.getNumPendingBlockFetches()
```

Comment:
Shouldn't the condition `executor.numRunningTasks == 0` avoid this situation already? If there are pending shuffle fetches, running tasks should be empty, right?

Reply:
No, these pending fetches means other executors are fetching from this executor.